shutdown.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"k8s.io/klog/v2"
+)
+
+// await for the signals and run the shutdown function
+func await(shutdownHook func()) {
```

Comment:
im not entierly sure of the value of having this separated out of the main function?

Making proper use of context and passing it to the relevant functions to stop them with a valid timeout should be done. instead of each server handling its own context. Context is generally used to communicate between domains/apis

Reply:
The timeout is handled in each part of the shutdown procedure.
The bus for instance needs to finish to process the messages it has in the queue before shutting down so a global timeout won't do it.

Secondly it's cleaner this way to separate it